### ISSUE TITLE
Turnaround time longer on Linux over conf.L2socket

### ISSUE THREAD

**Issue by hecrabbs** [secdev/scapy#3713]
### Brief description

Packets are taking ~0.1 seconds to be received over conf.L2socket on Linux.

### Scapy version

2.4.5

### Python version

3.6.9

### Operating system

Ubuntu 18.04.6

### How to reproduce

Running the attached echo_demo.py file (zipped) which uses the [SatCat5 python API](https://github.com/the-aerospace-corporation/satcat5/tree/main/src/python), will print the turnaround time it takes to send and receive a message.  On the Linux machine (Ubuntu 18.04.6) the turnaround time is consistently ~0.1 seconds, but Wireshark shows that the packet is actually received in less than 0.001 seconds.  Running it on a Windows machine (with Npcap 1.60) shows similar results to the Wireshark output.  

[echo_demo.zip](https://github.com/secdev/scapy/files/9305068/echo_demo.zip)

### Actual result

![Actual_Result](https://user-images.githubusercontent.com/58491968/184039072-d29852a0-a8b0-4cce-b10e-791c7714fcd1.png)

### Expected result

![Expected_Result](https://user-images.githubusercontent.com/58491968/184039265-ebfafdd9-b226-4d9a-bad9-44df97942c2d.png)

**Comment by gpotter2** [comment]
Hi, do you have the same issue if you enable
`conf.use_pcap = True`
on linux?

**Comment by hecrabbs** [comment]
Yes these results are with it enabled.  It is enabled on line 33 of satcat5_eth.py

**Comment by gpotter2** [comment]
How about with it disabled?

**Comment by hecrabbs** [comment]
Disabling `conf.use_pcap` and changing the `.nonblock_recv()` to `.recv()` gave these results:

![image](https://user-images.githubusercontent.com/58491968/184263266-3772be7d-e826-440b-ae1f-2bfbb1e0d60f.png)

**Comment by hecrabbs** [comment]
So it's better, but still not nearly as fast as it should be?

**Comment by gpotter2** [comment]
Yeah you're right. I'll have to check if I can reproduce..

Can you retry with master to be sure? rather than 2.4.5

**Comment by hecrabbs** [comment]
I tried with master and it was the same result

**Comment by hecrabbs** [comment]
Retrying with master gave the same results.

Could the issue be related to this?
https://www.tcpdump.org/faq.html#q15
> In immediate mode libpcap receives captured packets from the OS kernel one at a time, which results in a higher CPU overhead and a more likely packet loss; in buffered mode it receives multiple packets at once, but only after the buffer is full or its timeout has expired, which might take much longer than expected

**Comment by hecrabbs** [comment]
@gpotter2  any update on reproducing it?
